### DISCUSSION TITLE
Reviung39mkII?

### DISCUSSION THREAD

**Discussion by fliberdygibits** [manna-harbour/miryoku#113] [Q&A] [answered]
Before I dig into this myself I thought I would ask.  The link said requests for unsupported keyboards here.

**Comment by fliberdygibits** [comment]
> Before I dig into this myself I thought I would ask. The link said requests for unsupported keyboards here.

My apologies, I mis-read that all.  That's a request for unchecked keyboards... the 39 is out of tree.  Thank you:)

**Comment by manna-harbour** [comment, marked as answer]
Yes, reviung39 is already included. But due to the lack of thumb keys it might not be usable. This is the mapping: https://github.com/manna-harbour/miryoku_zmk/blob/master/miryoku/mapping/39/reviung39.h. The missing thumb keys are just moved to the outer pinkie columns. There's a sample build in this run: https://github.com/manna-harbour/miryoku_zmk/actions/runs/2426618968.

**Comment by fliberdygibits** [comment]
I tried the pre-compiled and it's not cooperating... as you indicated it's
the bottom row keys that are mis-maping.  I'll take a look at the code but
I'm VERY new to this so it'll be slow.  Thank you, your time is much
apprecaited:)

On Thu, Aug 4, 2022 at 5:55 PM Manna Harbour ***@***.***>
wrote:

> Yes, reviung39 is already included. But due to the lack of thumb keys it
> might not be usable. This is the mapping:
> https://github.com/manna-harbour/miryoku_zmk/blob/master/miryoku/mapping/39/reviung39.h.
> The missing thumb keys are just moved to the outer pinkie columns. There's
> a sample build in this run:
> https://github.com/manna-harbour/miryoku_zmk/actions/runs/2426618968.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/manna-harbour/miryoku/discussions/113#discussioncomment-3329569>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOWBYVQLWJHRRZJDHGEVOXDVXRRBLANCNFSM55UG5MEQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
I'll be more willing to encourage thinking outside the box when I see some
evidence of thinking inside the box.
-Terry Pratchett

**Reply by manna-harbour** [reply]
`K32`-`K37` here https://github.com/manna-harbour/miryoku_zmk/blob/095fd13c5c003008d33d001a81702ae1f9d7ba74/miryoku/mapping/39/reviung39.h#L12 are, in order, the left tertiary, primary and secondary and the right secondary, primary, and tertiary thumb keys.

You can see where they end up on reviung39 here https://github.com/manna-harbour/miryoku_zmk/blob/095fd13c5c003008d33d001a81702ae1f9d7ba74/miryoku/mapping/39/reviung39.h#L14-L17.
